helix-term/src/commands.rs

```diff
@@ -418,25 +421,41 @@ fn goto_line_end(cx: &mut Context) {
         let pos = graphemes::prev_grapheme_boundary(text, line_end_char_index(&text, line))
             .max(line_start);
 
-        range.put_cursor(text, pos, doc.mode == Mode::Select)
+        range.put_cursor(text, pos, always_extend || doc.mode == Mode::Select)
     });
     doc.set_selection(view.id, selection);
 }
 
-fn goto_line_end_newline(cx: &mut Context) {
+fn goto_line_end(cx: &mut Context) {
+    goto_line_end_impl(cx, false)
+}
+
+fn extend_to_line_end(cx: &mut Context) {
+    goto_line_end_impl(cx, true)
+}
+
+fn goto_line_end_newline_impl(cx: &mut Context, always_extend: bool) {
     let (view, doc) = current!(cx.editor);
     let text = doc.text().slice(..);
 
     let selection = doc.selection(view.id).clone().transform(|range| {
         let line = range.cursor_line(text);
         let pos = line_end_char_index(&text, line);
 
-        range.put_cursor(text, pos, doc.mode == Mode::Select)
+        range.put_cursor(text, pos, always_extend || doc.mode == Mode::Select)
```

Comment:
I'd rather see this parameter called `extend`, then you can map to the new `extend_` commands inside the keymap https://github.com/helix-editor/helix/blob/918bde0b05ceef596283652a1c4c922f35f3921d/helix-term/src/keymap.rs#L577

(So two sets of commands, one that always moves, one that always extends)

Reply:
Even better than a bool would be if we reused this enum: https://github.com/helix-editor/helix/blob/3fda350494449bf5909a0b47f9e2f593fbe615ad/helix-core/src/movement.rs#L21-L25